src/Symfony/Component/Security/Core/User/User.php

```diff
@@ -18,7 +18,7 @@
  *
  * @author Fabien Potencier <fabien@symfony.com>
  */
-final class User implements AdvancedUserInterface
+final class User implements UserInterface, EquatableInterface
```

Comment:
> I just need to find a way to not trigger a deprecation for that.

In the user checker? I guess you could check the classname before triggering, and if it is the core user class, don't trigger.